src/Bridge/Symfony/Routing/ApiLoader.php

```diff
@@ -55,14 +59,18 @@ public function load($data, $type = null)
 
         foreach ($this->resourceNameCollectionFactory->create() as $resourceClass) {
             $resourceMetadata = $this->resourceMetadataFactory->create($resourceClass);
-            $normalizedShortName = Inflector::pluralize(Inflector::tableize($resourceMetadata->getShortName()));
+            $resourceShortName = $resourceMetadata->getShortName();
+
+            if (null === $resourceShortName) {
+                throw new InvalidResourceException(sprintf('Resource "%s" has no short name defined.', $resourceClass));
```

Comment:
As long as the [`ShortNameResourceMetadataFactory`](https://github.com/api-platform/core/blob/master/src/Metadata/Resource/Factory/ShortNameResourceMetadataFactory.php) is called before we hit `ApiLoader` the short name will never be null. IMO you should remove this test and reflect the `shortName` in the test that failed. Still, the test would have to be valid for [this error](https://github.com/polc/api-platform-core/blob/43ac243d3849b05282f31e2cc27ef4b188579031/src/Bridge/Symfony/Routing/ApiLoader.php#L107).